File: connection.py
from requests import get
from log import get_logger
from time import sleep

logger = get_logger(__name__)

def _is_connected():
    try:
        get("https://1.1.1.1")
        return True
    except:
        logger.warning("Failed to connect to the internet.")
        return False


def wait_for_connection():
    while not _is_connected():
        sleep(1)

File: internal/connection/connection.go
package connection

import (
	"log/slog"
	"net/http"
	"time"
)

func isConnected() bool {
	client := &http.Client{Timeout: 5 * time.Second}
	resp, err := client.Get("https://1.1.1.1")
	if err != nil {
		slog.Warn("failed to connect to the internet")
		return false
	}
	resp.Body.Close()
	return true
}

func WaitForConnection() {
	for !isConnected() {
		time.Sleep(time.Second)
	}
}
